crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -110,8 +110,14 @@ impl SlidingSyncList {
     ///
     /// There's no guarantee of ordering between items emitted by this stream
     /// and those emitted by other streams exposed on this structure.
-    pub fn state_stream(&self) -> impl Stream<Item = SlidingSyncState> {
-        Observable::subscribe(&self.inner.state.read().unwrap())
+    pub fn state_stream(
+        &self,
+    ) -> (SlidingSyncListLoadingState, impl Stream<Item = SlidingSyncListLoadingState>) {
```

Comment:
Are you trying to avoid making `eyeball` a part of the public API? If not, you could just return a `Subscriber` which also allows getting the current value.

Also shouldn't the method documentation be updated?

Reply:
I'm trying to be consistent with other methods. Ideally, I would like to return a `Subscriber` for everything, but I keep that for another clean up.

I'll update the documentation :-p.